docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,559 @@
+    ports:
+      - name: port-5432
+        port: 5432
+        targetPort: 5432
+
+
+- name: create app route
+  openshift_v1_route:
+    name: manageiq-app
+    namespace: '{{ namespace }}'
+    spec_port_target_port: port-443
+    spec_tls_termination: passthrough
+    labels:
+      app: manageiq-apb
+      service: manageiq-app
+    to_name: manageiq-app
+
+
+- name: create app deployment config
+  openshift_v1_deployment_config:
+    name: manageiq-app
+    namespace: '{{ namespace }}'
+    labels:
+      app: manageiq-apb
+      service: manageiq-app
+    replicas: 1
+    selector:
+      app: manageiq-apb
+      service: manageiq-app
+    spec_template_metadata_labels:
+      app: manageiq-apb
+      service: manageiq-app
+    containers:
+      - env:
+          - name: MIQ_MEMCACHED_PORT_11211_TCP
+            value: "{{ manageiq_memcached_port_11211_tcp | string }}"
+          - name: DATABASE_SERVICE_NAME
+            value: "{{ database_service_name | string }}"
+          - name: MIQ_POSTGRESQL_PORT_5432_TCP_ADDR
+            value: "{{ manageiq_postgresql_5432_tcp_addr }}"
+          - name: MIQ_POSTGRESQL_PORT_5432_TCP
+            value: "{{ manageiq_postgresql_port_5432_tcp | string}}"
+          - name: DATABASE_REGION
+            value: "{{ database_region | string }}"
+          - name: MIQ_POSTGRESQL_SERVICE_HOST
+            value: "{{ manageiq_postgresql_service_host }}"
+          - name: MIQ_MEMCACHED_PORT_11211_TCP_ADDR
+            value: "{{ manageiq_memcached_port_11211_tcp_addr }}"
+          - name: POSTGRESQL_PASSWORD
+            value: "{{ postgresql_password }}"
+          - name: MEMCACHED_SERVICE_NAME
+            value: "{{ memcached_service_name }}"
+          - name: APPLICATION_INIT_DELAY
+            value: "{{ application_init_delay | string}}"
+          - name: MIQ_MEMCACHED_PORT
+            value: "{{ manageiq_memcached_port | string}}"
+          - name: MIQ_POSTGRESQL_PORT
+            value: "{{ manageiq_postgresql_port | string }}"
+          - name: MIQ_MEMCACHED_SERVICE_HOST
+            value: "{{ manageiq_memcached_service_host }}"
+          - name: POSTGRESQL_USER
+            value: "{{ postgresql_user }}"
+          - name: POSTGRESQL_DATABASE
+            value: "{{ postgresql_database }}"
+        image: docker.io/manageiq/manageiq-pods:app-latest-fine
+        name: manageiq-app
+        ports:
+        - container_port: 80
+          protocol: TCP
+        - container_port: 443
+          protocol: TCP
+        security_context:
+          privileged: true
+        termination_message_path: /dev/termination-log
+        volume_mounts:
+        - mount_path: /persistent
+          name: manageiq-app
+        working_dir: /
+    volumes:
+    - name: manageiq-app
+      persistent_volume_claim:
+        claim_name: manageiq-app
+    test: false
+    triggers:
+    - type: ConfigChange
+
+
+
+- name: create manageiq-memcached deployment config
+  openshift_v1_deployment_config:
```

Comment:
So in an APB world, needing non-core modules such as the above in your container would somewhat lead back to a file copying problem. Hopefully we could get something more solid into ansible core for this kind of kube API so that's no longer necessary. Perhaps there's a way to layer external modules in when building an APB as well.

Reply:
So this was something that I had to ask @fabianvf about, where were these modules coming from when looking at some of their APB examples. When building the docker image, if you build `FROM ansibleplaybookbundle/apb-base` then you have access to these. If we make it a requirement that all the APB images source from this, then we would have a single location for APBs to maintain these modules from...

Not sure if that addresses your concern at all?

Reply:
Ah they're in there by default? Interesting. No major concern here, and I hope there'll be rich kube modules in core before too long as well.